### PR TITLE
fix: dataset diff

### PR DESCRIPTION
**Pull request by MillenniumFalconMechanic**
## Reason for Change

#6508

## Changes

- Updated published counterpart (used when diff'ing revisions) to be built from its own datasets and not the revision's datasets.

## Testing steps

- Tested manually in rdev.
- ~~Added e2e test~~. I am experiencing flakiness in the e2e test and have created [Add e2e test exercising dataset diff functionality](https://github.com/chanzuckerberg/single-cell-data-portal/issues/6561) to address this separately.

### CODE REVIEW

File: frontend/src/common/queries/collections.ts [hunk 1]
```diff
@@ -153,8 +153,8 @@ function fetchCollection() {
 
     const url = apiTemplateToUrl(API_URL + API.COLLECTION, { id });
 
-    let response = await fetch(url, DEFAULT_FETCH_OPTIONS);
-    let json = await response.json();
+    const response = await fetch(url, DEFAULT_FETCH_OPTIONS);
+    const json = await response.json();
```
Comment (MillenniumFalconMechanic) [inline frontend/src/common/queries/collections.ts:157]: Updating `response` and `json` to `const` to help identify them as different objects from the published counterpart's response (`publishedCounterpartResponse`) and json (`publishedCounterpartJSON`) objects below.

File: frontend/src/common/queries/collections.ts [hunk 2]
```diff
@@ -164,16 +164,7 @@ function fetchCollection() {
...
-      (dataset: DatasetResponse) => {
-        // It's possible for a dataset not to have tissues defined during
-        // upload; protect with [].
-        const tissue = (dataset.tissue ?? []).map((tissue) =>
-          createTaggedTissueOntology(tissue)
-        );
-        return { ...dataset, tissue };
-      }
-    );
+    const datasets: Dataset[] = createTaggedTissueOntologies(json.datasets);
```
Comment (MillenniumFalconMechanic) [inline frontend/src/common/queries/collections.ts:167]: Moved functionality to function below for re-use.

File: frontend/src/common/queries/collections.ts [hunk 3]
```diff
@@ -182,17 +173,27 @@ function fetchCollection() {
...
+      const publishedCounterpartResponse = await fetch(
+        publicCollectionURL,
+        DEFAULT_FETCH_OPTIONS
+      );
+      const publishedCounterpartJSON =
+        await publishedCounterpartResponse.json();
+
+      // Convert tissue ontology objects to core ontology objects.
+      const publishedCounterpartDatasets: Dataset[] =
+        createTaggedTissueOntologies(publishedCounterpartJSON.datasets);
```
Comment (MillenniumFalconMechanic) [inline frontend/src/common/queries/collections.ts:191]: Renamed variables here to clearly indicated they are associated with the published counterpart and not the collection (related to `let` to `const` update above).

File: frontend/src/common/queries/collections.ts [hunk 4]
```diff
@@ -182,17 +173,27 @@ function fetchCollection() {
...
         id: collection.revision_of,
       });
-      response = await fetch(publicCollectionURL, DEFAULT_FETCH_OPTIONS);
-      json = await response.json();
+      const publishedCounterpartResponse = await fetch(
+        publicCollectionURL,
+        DEFAULT_FETCH_OPTIONS
+      );
+      const publishedCounterpartJSON =
+        await publishedCounterpartResponse.json();
```
Comment (MillenniumFalconMechanic) [inline frontend/src/common/queries/collections.ts:187]: Renamed variables here to clearly indicated they are associated with the published counterpart and not the collection (related to `let` to `const` update above).

**Review by tihuan — APPROVED** [review]
Love the refactor and fix 🙏 Thanks so much, Mim!!

**Pull request merged**